src/native/mac/qgsmacnative.mm

```diff
@@ -18,10 +18,43 @@
 #include "qgsmacnative.h"
 
 #include <Cocoa/Cocoa.h>
+#include <QtMacExtras/QtMac>
+
 #include <QString>
+#include <QPixmap>
+
+
+@interface QgsUserNotificationCenterDelegate : NSObject <NSUserNotificationCenterDelegate>
+@end
+
+@implementation QgsUserNotificationCenterDelegate
+
+- ( BOOL )userNotificationCenter:( NSUserNotificationCenter * )center shouldPresentNotification:( NSUserNotification * )notification
+{
+#pragma unused (notification)
+#pragma unused (center)
+  return YES;
+}
+
+@end
+
+class QgsMacNative::QgsUserNotificationCenter
+{
+  public:
+    QgsUserNotificationCenterDelegate *_qgsUserNotificationCenter;
```

Comment:
Do you need to (can you?) initialize here? Is this a unique_ptr candidate?

Reply:
nope, I've tried it but you cannot use unique_ptr on the objective c implementation

![image](https://user-images.githubusercontent.com/127259/44443388-2235ab00-a5e0-11e8-9be4-d4491fe6eab1.png)